aiida/cmdline/commands/cmd_code.py

```diff
@@ -175,7 +183,7 @@ def code_duplicate(ctx, code, non_interactive, **kwargs):
 
     try:
         new_code.store()
```

Comment:
I think we need validate_label_uniqueness after the code is created and stored in DB. Since for remote code, it loads code to check when the exception is raised.

Reply:
Not sure I understand what you mean here. I think the uniqueness validation is done automatically on the database level through uniqueness constraints no?

Reply:
I think that the `duplicate` command is broken now though. It also needs to be made a dynamic group. But not sure how we can do this without breaking the original one. Just like we need to make the new `setup` command `create` we technically might need a different name for the updated `duplicate` as well, but not sure what

Reply:
> Not sure I understand what you mean here. I think the uniqueness validation is done automatically on the database level through uniqueness constraints no?

`validata_label_uniqueness` check duplicate by load_code, if the new code is store after check the check will pass anyway and there will be two identical codes (not tested but only look at the logic here I think it is the case, no? maybe I miss something?).
https://github.com/aiidateam/aiida-core/blob/25309f74be97e5ab176de9a71b418dedb258ba4d/aiida/cmdline/params/options/commands/code.py#L26

Reply:
> But not sure how we can do this without breaking the original one. Just like we need to make the new setup command create we technically might need a different name for the updated duplicate as well, but not sure what

I see. I can image two use cases of using `verdi code duplicate`:

1. The original code is invalid and the user want to use most of options without setup a code from scratch
2. The original code is still under use but the user wants to set a new one with some attributes modified.

For 1, I think the original code can be relabeled to suffix with <code-name>-<unique-key>-deprecate and hide.
For 2, I think the new code can be asked to have a new label without touching the original code. 

It can be hard to implement this logic clearly I guess?

Reply:
My point was more that the current interface of `verdi code duplicate` only works with the now deprecated `LegacyCode`. The new command should have dynamic options, based on the plugin type of the code that is being duplicated. So just like `verdi code setup` is deprecated and is replaced by `verdi code create`, `verdi code duplicate` should technically be deprecated and be replaced by another command.